app/IATI/Services/Activity/ResultService.php

```diff
@@ -165,6 +165,61 @@ public function getResultWithIndicatorAndPeriod($resultId, $activityId): ?Model
         return $this->resultRepository->getResultWithIndicatorAndPeriod($resultId, $activityId);
     }
 
+    /**
+     * Checks if result indicator has ref code.
+     *
+     * @param $resultId
+     *
+     * @return bool
+     */
+    public function indicatorHasRefCode($resultId): bool
+    {
+        $result = $this->resultRepository->getResultWithIndicator($resultId);
+
+        if (!empty($result['indicators'])) {
+            $indicators = $result['indicators'];
+
+            foreach ($indicators as $item) {
+                $indicator = $item['indicator'];
+                $refs = $indicator['reference'];
```

Comment:
Can $refs be directly assigned as $item['indicator']['reference'] or Arr::get($item,'indicator.reference')  since variable $indicator is not used elsewhere?

Reply:
Can be but code readability becomes easy